include/hwinfo/network.h

```diff
@@ -0,0 +1,35 @@
+// Copyright MRsoymilk
+// Author MRsoymilk <codermrsoymilk@gmail.com>
+
+#pragma once
+
+#include <hwinfo/platform.h>
+
+#include <string>
+#include <vector>
+
+namespace hwinfo {
+
+class Network {
+ public:
+  friend std::vector<Network> getAllNetworks();
+
+public:
+  ~Network() = default;
+
+  HWI_NODISCARD const std::string& interfaceIndex() const;
```

Comment:
The new HWinfo style is to return std::string instead of const std::string& because of copy elision